UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs

```diff
@@ -492,6 +509,13 @@ public void RollRandomCharacter()
 			eyeDropdown.BodyPartColour = UnityEngine.Random.ColorHSV();
 			eyeDropdown.RequestColourPicker();
 
+			//Randomize hair and facial hair.
+			int randomHair = UnityEngine.Random.Range(0, hairList.OptionalSprites.Count - 1);
+			hairList.Dropdown.value = randomHair;
+			hairList.ItemChange(randomHair);
+			hairList.BodyPartColour = UnityEngine.Random.ColorHSV();
+			hairList.RequestColourPicker();
```

Comment:
Another way that might be good is having the component handle the randomisation itself , you get a list when generating of BodyPartCustomisationBase  and then you just loop through and call randomise On each one